### PR TITLE
ref(workflow_engine): Update the Action Model

### PR DESCRIPTION
**Pull request by saponifi3d**
## Description
Chatted with Colleen about `AlertRuleTriggerActions` => `Actions` and found a few fields that are missing in the model, that shouldn't be part of the `data` blob. Adding these to the Action table for now.

Given the Notification Platform work, we will probably want to move these to a new table as part of that project -- so these are all marked as "LEGACY" to denote the fields that I feel like we'll want to eventually remove.

[More info the spec]( https://www.notion.so/Alerts-Create-Issues-Metric-Alert-Migration-1198b10e4b5d80e29cdffc191de502c7?d=12d8b10e4b5d80cc93af001c5d42edac&pvs=4#11c8b10e4b5d80658f82c4f8a231fceb)

### CODE REVIEW

File: src/sentry/workflow_engine/models/action.py [hunk 1]
```diff
@@ -26,5 +29,19 @@ class Type(models.TextChoices):
     """
     required = models.BooleanField(default=False)
 
+    # The type field is used to denote the type of action we want to trigger
     type = models.TextField(choices=Type.choices)
     data = models.JSONField(default=dict)
+
+    # LEGACY: The integration_id is used to map the integration_id found in the AlertRuleTriggerAction
+    # This allows us to map the way we're saving the notification channels to the action.
+    integration_id = models.BigIntegerField(null=True)
```
Comment (ceorourke) [inline src/sentry/workflow_engine/models/action.py:38]: [Here's](https://github.com/getsentry/sentry/blob/master/src/sentry/models/notificationaction.py#L152) the existing implementation, is there a reason not to make this a FK? Same question for `target_type`, is there a reason to not have it be a smallint mapping to the enum?

Comment (saponifi3d) [reply src/sentry/workflow_engine/models/action.py:38]: 👍 i'll update, sorry, i put this together between meetings 😓

File: src/sentry/workflow_engine/models/action.py [hunk 2]
```diff
@@ -26,5 +31,21 @@ class Type(models.TextChoices):
...
+        "sentry.Integration", blank=True, null=True, on_delete="CASCADE"
+    )
+
+    # LEGACY: The target_display is used to display the target's name in notifications
+    target_display = models.TextField(null=True)
+
+    # LEGACY: The target_identifier is used to target the user / team / org that notifications are being sent to
+    target_identifier = models.TextField(null=True)
+
+    # LEGACY: This is used to denote if the Notification being sent is a Slack notification etc
```
Comment (ceorourke) [inline src/sentry/workflow_engine/models/action.py:50]: nit: this is to show whether it's going to a user, team, sentry app, etc.

**Review by ceorourke — APPROVED** [review]
lgtm based on what we discussed yesterday, best to wait for a migration owner to merge though

**Review by markstory — APPROVED** [review]
Schema changes look good to me.

**Review by wedamija — APPROVED** [review]
This makes sense to me, and migration looks good too

**Pull request merged**